os/se/ameba/security_ameba_wrapper_tz.c

```diff
@@ -0,0 +1,791 @@
+{
+	AWRAP_ENTER;
+	int ret;
+	inout_struc in_output;
+
+	key_idx = rtl_index_convt(key_idx);
+	if (key_idx >= KEY_STORAGE_INDEX_MAX) {	/* Index Out of Range */
+		return HAL_INVALID_SLOT_RANGE;
+	}
+
+	in_output.input_data = enc_data;
+	in_output.output_data = dec_data;
+	ret = ameba_hal_rsa_decrypt(&in_output, rsa_mode, key_idx);
+
+	if (ret != HAL_SUCCESS) {
+		sedbg("RTL SE failed (%zu)\n", ret);
+	}
+	return ret;
+}
+
+int se_ameba_hal_write_storage(uint32_t ss_idx, hal_data *data)
+{
+	AWRAP_ENTER;
+	int ret;
+
+	if ((ss_idx == 0) || (ss_idx > SEC_STORE_MAX_SLOT)) {	/* Index Out of Range */
+		return HAL_INVALID_SLOT_RANGE;
+	}
+
+	if (data->data_len > (SECTOR_SIZE * 2)) {
+		/* If data len more than 2 Secotr (8KB) */
+		return HAL_FAIL;
+	}
+	else {
+		/* Only Slot 1 can write 8KB, other is less then 4KB */
+		if (ss_idx != 1) {
+			/* If data len more Sector - Tag size and not Slot 1 */
+			if ((data->data_len > (SECTOR_SIZE - 32)) && (ss_idx != 1)) {
+				return HAL_FAIL;
+			}
+		}
+	}
+
+	ret = ameba_hal_write_storage(ss_idx, data);
+
+	if (ret != HAL_SUCCESS) {
+		sedbg("RTL SE failed (%zu)\n", ret);
+	}
+	return ret;
+}
+
+int se_ameba_hal_read_storage(uint32_t ss_idx, hal_data *data)
+{
+	AWRAP_ENTER;
+	int ret;
+
+	if ((ss_idx == 0) || (ss_idx > SEC_STORE_MAX_SLOT)) {	/* Index Out of Range */
+		return HAL_INVALID_SLOT_RANGE;
+	}
+
+	ret = ameba_hal_read_storage(ss_idx, data);
+
+	if (ret != HAL_SUCCESS) {
+		sedbg("RTL SE failed (%zu)\n", ret);
+	}
+	return ret;
+}
+
+int se_ameba_hal_delete_storage(uint32_t ss_idx)
+{
+	AWRAP_ENTER;
+	int ret;
+
+	if ((ss_idx == 0) || (ss_idx > SEC_STORE_MAX_SLOT)) {	/* Index Out of Range */
+		return HAL_INVALID_SLOT_RANGE;
+	}
+
+	ret = ameba_hal_delete_storage(ss_idx);
+
+	if (ret != HAL_SUCCESS) {
+		sedbg("RTL SE failed (%zu)\n", ret);
+	}
+	return ret;
+}
+
+
+static struct sec_ops_s g_ameba_ops_s = {
+	se_ameba_hal_init,
+	se_ameba_hal_deinit,
+	se_ameba_hal_free_data,
+	se_ameba_hal_get_status,
+	se_ameba_hal_set_key,
+	se_ameba_hal_get_key,
+	se_ameba_hal_remove_key,
+	se_ameba_hal_generate_key,
+	se_ameba_hal_generate_random,
+	se_ameba_hal_get_hash,
+	se_ameba_hal_get_hmac,
+	se_ameba_hal_rsa_sign_md,
+	se_ameba_hal_rsa_verify_md,
+	se_ameba_hal_ecdsa_sign_md,
+	se_ameba_hal_ecdsa_verify_md,
+	se_ameba_hal_dh_generate_param,
+	se_ameba_hal_dh_compute_shared_secret,
+	se_ameba_hal_ecdh_compute_shared_secret,
+	se_ameba_hal_set_certificate,
+	se_ameba_hal_get_certificate,
+	se_ameba_hal_remove_certificate,
+	se_ameba_hal_get_factory_key,
+	se_ameba_hal_get_factory_cert,
+	se_ameba_hal_get_factory_data,
+	se_ameba_hal_aes_encrypt,
+	se_ameba_hal_aes_decrypt,
+	se_ameba_hal_rsa_encrypt,
+	se_ameba_hal_rsa_decrypt,
+	se_ameba_hal_write_storage,
+	se_ameba_hal_read_storage,
+	se_ameba_hal_delete_storage,
+};
+
+static struct sec_lowerhalf_s g_ameba_lower = {&g_ameba_ops_s, NULL};
+
+#define SECLINK_PATH "/dev/seclink"
```

Comment:
Please refer to this. You can just include the proper header.
https://github.com/Samsung/TizenRT/pull/4628

Reply:
Thank you, will change accordingly.